subsys/net/ip/ipv6_nbr.c

```diff
@@ -1340,15 +1339,15 @@ static enum net_verdict handle_ns_input(struct net_pkt *pkt,
 send_na:
 	if (src_lladdr.len) {
 		if (!net_ipv6_nbr_add(net_pkt_iface(pkt),
-				      &NET_IPV6_HDR(pkt)->src,
+				      &ip_hdr->src,
```

Comment:
good to get rid of this left-over

Reply:
good to get rid of the address swapping ;-)